crates/snapbox/src/data.rs

```diff
@@ -110,10 +107,10 @@ impl Data {
     /// Return the underlying `str`
     ///
     /// Note: this will not inspect binary data for being a valid `str`.
-    pub fn as_str(&self) -> Option<&str> {
+    pub fn try_string(&self) -> Option<String> {
```

Comment:
> change as_str to try_string and update its return to be Option<String>
> - This was to allow for different data formats that don't have an easy way to be converted to &str

This is in the checklist as

> as_str should be renamed to try_string and it should output Option<String>. This is because some data formats cannot be turned into a &str without borrowing issues

What neither captures is what is the intention behind `as_str` / `try_string`.  What role is it filling?  I can go through and refresh myself on that but the PR should do that.

Reply:
I updated the PR description to provide what `as_str` is for and why changing it is needed